components/org.wso2.carbon.identity.workflow.engine/src/main/java/org/wso2/carbon/identity/workflow/engine/ApprovalTaskServiceImpl.java

```diff
@@ -219,9 +218,13 @@ public void updateApprovalTaskStatus(String approvalTaskId, StateDTO nextState)
     public void addApprovalTasksForWorkflowRequest(WorkflowRequest workflowRequest, List<Parameter> parameterList)
             throws WorkflowEngineException {
 
+        if (CollectionUtils.isEmpty(parameterList)) {
+            return;
+        }
+
         String tenantDomain = IdentityTenantUtil.getTenantDomain(workflowRequest.getTenantId());
         String workflowRequestId = getWorkflowRequestId(workflowRequest);
-        String workflowId = getWorkflowId(workflowRequest);
+        String workflowId = parameterList.get(0).getWorkflowId();
```

Comment:
Let's add a comment on the reason.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-workflow/pull/19/commits/11966347d5e1dfcf9e43ee56627da72e5fc32b6d